packages/editor/src/EditorChildWindow.tsx

```diff
@@ -69,25 +67,32 @@ export function EasyblocksCanvas({
   }, []);
 
   useEffect(() => {
-    window.parent.editorWindowAPI.onUpdate = () => {
-      // Force re-render when child gets info from parent that data changed
-      forceRerender();
-    };
-  });
+    window.parent.postMessage(
+      {
+        type: "@easyblocks-editor/canvas-loaded",
+      },
+      "*"
+    );
+  }, []);
 
-  const shouldNotRender = !enabled || !meta || !compiled || !externalData;
+  const { meta, compiled, externalData, formValues, definitions } =
+    useEasyblocksCanvasContext();
+
+  const shouldNotRender =
+    !enabled ||
+    !meta ||
+    !compiled ||
+    !externalData ||
+    !formValues ||
+    !definitions;
```

Comment:
This condition has become monster (I know it was like that before). If we simplify the output of `useEasyblocksCanvasContext` so that it can be `null` then the only things we'll need to check is `enabled`, `externalData` and if context is not null.